rust/datafusion/src/execution/physical_plan/expressions.rs

```diff
@@ -51,3 +59,163 @@ impl PhysicalExpr for Column {
         Ok(batch.column(self.index).clone())
     }
 }
+
+/// Create a column expression
+pub fn col(i: usize) -> Arc<dyn PhysicalExpr> {
+    Arc::new(Column::new(i))
+}
+
+/// SUM aggregate expression
+pub struct Sum {
+    expr: Arc<dyn PhysicalExpr>,
+}
+
+impl Sum {
+    /// Create a new SUM aggregate function
+    pub fn new(expr: Arc<dyn PhysicalExpr>) -> Self {
+        Sum { expr }
+    }
+}
+
+impl AggregateExpr for Sum {
+    fn name(&self) -> String {
+        "SUM".to_string()
+    }
+
+    fn data_type(&self, input_schema: &Schema) -> Result<DataType> {
+        match self.expr.data_type(input_schema)? {
+            DataType::Int8 | DataType::Int16 | DataType::Int32 | DataType::Int64 => {
+                Ok(DataType::Int64)
+            }
+            DataType::UInt8 | DataType::UInt16 | DataType::UInt32 | DataType::UInt64 => {
+                Ok(DataType::UInt64)
+            }
+            other => Ok(other.clone()),
```

Comment:
Can we be explicit here about what else is allowed?  i.e. floats should pass through but everything else should be an `Err` right?